README.md

```diff
@@ -1,14 +1,615 @@
+```
+
+The second argument to `rooms.get` is a `RoomOptions` argument, which tells the Chat SDK what features you would like your room to use and
+how they should be configured.
+
+For example, you can set the timeout between keystrokes for typing events as part of the room options. Sensible defaults for each of the
+features are provided for your convenience:
+
+- A typing timeout (time of inactivity before typing stops) of 5 seconds.
+- Entry into, and subscription to, presence.
+
+The defaults options for each feature may be
+viewed [here](https://github.com/ably/ably-chat-js/blob/main/chat-android/src/main/java/com/ably/chat/RoomOptions.kt).
+
+In order to use the same room but with different options, you must first `release` the room before requesting an instance with the changed
+options (see below for more information on releasing rooms).
+
+Note that:
+
+- If a `release` call is currently in progress for the room (see below), then a call to `get` will wait for that to resolve before resolving
+  itself.
+- If a `get` call is currently in progress for the room and `release` is called, the `get` call will reject.
+
+### Attaching to a room
+
+To start receiving events on a room, it must first be attached. This can be done using the `attach` method.
+
+```kotlin
+// Add a listener so it's ready at attach time (see below for more information on listeners)
+room.messages.subscribe { message: Message ->
+    println(message.toString())
+}
+
+room.attach()
+```
+
+### Detaching from a room
+
+To stop receiving events on a room, it must be detached, which can be achieved by using the `detach` method.
+
+```kotlin
+room.detach()
+```
+
+Note: This does not remove any event listeners you have registered and they will begin to receive events again in the
+event that the room is re-attached.
+
+### Releasing a room
+
+Depending on your application, you may have multiple rooms that come and go over time (e.g. if you are running 1:1 support chat). When you
+are completely finished with a room, you may `release` it which allows the underlying resources to be collected.
+
+```kotlin
+rooms.release("basketball-stream")
+```
+
+Once `release` is called, the room will become unusable and you will need to get a new instance using `rooms.get` should you wish to
+re-start the room.
+
+Note that releasing a room may be optional for many applications.
+
+### Monitoring room status
+
+Monitoring the status of the room is key to a number of common chat features. For example, you might want to display a warning when the room
+has become detached.
+
+### Current status of a room
+
+To get the current status, you can use the `status` property:
+
+```kotlin
+val roomStatus = room.status
+val roomError = room.error
+```
+
+### Listening to room status updates
+
+You can also subscribe to changes in the room status and be notified whenever they happen by registering a listener:
+
+```kotlin
+val subscription = room.onStatusChange { statusChange: RoomStatusChange ->
```

Comment:
Not for this PR to fix, but OOI did we consider using `StateFlows` for this purpose - I'd imagine they fit quite well with the status change paradigm?

Reply:
With current Api design, we have problem with unsubscribing to state flows . Since,  it's not deterministic in nature : ( 
Maybe we can expose flow as an extension method in the future alongside current public API

Reply:
Ah, fair enough - that makes sense, so it's like you call `unsubscribe` but there's no guarantees that that will happen immediately, or that you won't receive any other events in the time?

But that's reason enough to not do this now, I was just curious - thanks for the explanation. We can definitely consider in the future and I would agree that it's an extension to the current API, if we choose to do this :)